src/store/user/user-context.js

```diff
@@ -10,6 +11,24 @@ export const UserContextProvider = ({ children }) => {
   const [showMemberRoleUpdateModal, setShowMemberRoleUpdateModal] =
     useState(false);
   const [selectedMember, setSelectedMember] = useState(null);
+  const [superUserVerified, setSuperUserVerified] = useState(false);
+
+  const verifySuperUser = async () => {
+    setUser(null);
+    setIsSuperUser(false);
+    setSuperUserVerified(false);
+    try {
+      const { data } = await getUserSelf();
+      if (data) {
+        setUser(data);
+        setIsSuperUser(Boolean(data.roles?.super_user));
+        setSuperUserVerified(true);
```

Comment:
if both the `isSuperUser` and `superUserVerified` are getting settled to true at the same time then can't we just switch to one state instead of multiple cause I found it redundant?

Reply:
`isSuperUser` checks if a user is or is not a superUser but `superUserVerified` checks if the API call to verify super User is completed. So the next time we don't make an API call again to verify whether or not the user is a superUser.